### PR TITLE
Displays a FontAwesome Asterisk when a Property is Mandatory

### PR DESCRIPTION
**Pull request by jermnelson**
In a ResourceTemplate, for all `literals`, `lookups`, and `resource` properties, if *mandatory* is set to "true", then displays a red asterisk (using FontAwesome) to indicate the component is required.

Changed for the following Components:
- InputLiteral, 
- InputLookupQA,
- InputListLOC
- ButtonToolbar in ResourceTemplateForm.

 In the spoofed `MonographInstance` Resource Template, the *Instance Of*'s mandatory property is set to **true**:
screen shot 2019-01-23 at 4 59 51 pm

Which generates the following label with a red asterisk:
screen shot 2019-01-23 at 5 07 31 pm

This is work required to complete Issue #303. @astridu may want to look at adjust as needed for the final user interface.

### CODE REVIEW

**Comment by astridu** [comment]
The screenshot looks good!

File: src/components/editor/InputListLOC.jsx [hunk 1]
```diff
@@ -39,6 +41,12 @@ class
     this.props.handleSelectedChange(payload)
   }
 
+  mandatorySuperscript() {
+    if (JSON.parse(this.props.propertyTemplate.mandatory)) {
+      return <sup><FontAwesomeIcon className="asterick text-danger" icon={faAsterisk} /></sup>
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (jermnelson) [reply src/components/editor/InputListLOC.jsx]: Good suggestion! Just pushed up a new commit that moves the display to it's own `RequiredSuperscript` React component, `InputLiteral`, `InputListLOC`, and `InputLookupQA` now use this new component.

**Comment by jermnelson** [comment]
I also added a new ticket #311 for the actual form validation of mandatory elements.

File: src/components/editor/ResourceTemplateForm.jsx [hunk 2]
```diff
@@ -55,6 +58,12 @@ class ResourceTemplateForm extends Component {
     return buttons
   }
 
+  mandatorySuperscript = (propMandatory) => {
+    if (JSON.parse(propMandatory)) {
+      return <sup><FontAwesomeIcon className="asterick text-danger" icon={faAsterisk} /></sup>
```
Comment (jgreben) [inline src/components/editor/ResourceTemplateForm.jsx]: Why is this different than the child input components? Why not also use `RequiredSuperscript ` here? Is there a nuance I am not seeing?

**Pull request merged**